source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -666,4 +744,93 @@ namespace AdaptiveCards::Rendering::Uwp
         auto foundSvg = url.find("svg");
         return !(foundSvg == std::string::npos);
     }
+
+    void render_xaml::XamlBuilder::ParseXmlForHeightAndWidth(winrt::XmlDocument const& xmlDoc,
+                                                             winrt::SvgImageSource const& imageSource)
+    {
+        if (xmlDoc)
+        {
+            auto rootElement = xmlDoc.DocumentElement();
+            auto height = rootElement.GetAttribute(L"height");
+            auto width = rootElement.GetAttribute(L"width");
+
+            // We only need to set height or width, not both (fixes aspect ratio for person style)
+            bool isHeightSet = false;
+
+            if (!height.empty())
+            {
+                if (auto heightAsDouble = HStringToDouble(height))
+                {
+                    SetRasterizedPixelHeight(imageSource, heightAsDouble);
+                    isHeightSet = true;
+                }
+            }
+
+            if (!width.empty())
+            {
+                if (auto widthAsDouble = HStringToDouble(width))
+                {
+                    SetRasterizedPixelWidth(imageSource, widthAsDouble, isHeightSet);
+                }
+            }
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetRasterizedPixelHeight(winrt::SvgImageSource const imageSource,
```

Comment:
>IAsyncAction [](http://example.com/codeflow?start=11&length=12)

Nobody is waiting on this.  Consider using `winrt::fire_and_forget` instead.

Reply:
Although there are two SetRasterizedPixelHeight implementations in this class so it is really hard to tell who is calling what (not that any of them look at the return value).

Reply:
And now I'm even more confused.  They differ only by return type and a 3rd parameter (which is a default parameter so callers do not require it).

Reply:
The `SetRasterizedPixelHeight` implementation with `void` returns are only called from `BuildImage`, so we don't have to worry about switching threads - this is why I split the methods.

It also takes in an `ImageSource` and handles the `SvgImageSource` casting to keep `BuildImage` slightly less cluttered

Renaming to `SetRasterizedPixelHeight` and `SetRasterizedPixelHeightAsync` would make it more clear?

Reply:
Different names would certainly make it more clear.